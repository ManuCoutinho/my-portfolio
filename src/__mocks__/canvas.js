HTMLCanvasElement.prototype.getContext = () => {
  return {
    fillRect: jest.fn()
  }
}
