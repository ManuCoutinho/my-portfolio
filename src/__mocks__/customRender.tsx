import { ReactElement, FC, ReactNode } from 'react'
import { render } from '@testing-library/react'
import { RenderOptions } from '@testing-library/react/types'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
	return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
