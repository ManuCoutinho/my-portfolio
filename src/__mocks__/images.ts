export const mockImg = [
	{
		imgPng: '/img1.png',
		imgWebp: '/img1.webp',
		alt: 'img1',
		id: 'img1'
	},
	{
		imgPng: '/img2.png',
		imgWebp: '/img2.webp',
		alt: 'img2',
		id: 'img2'
	}
]
