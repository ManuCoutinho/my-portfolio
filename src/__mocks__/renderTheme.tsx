import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
