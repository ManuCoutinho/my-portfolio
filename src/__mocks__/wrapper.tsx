/* eslint-disable react/display-name */
import { ThemeContextProvider } from 'contexts/ThemeContext'
import { DataProviderMock } from '__mocks__/providers/DataContext'

export const customWrapper = () => {
	return ({ children }: { children: React.ReactNode }) => (
		<ThemeContextProvider>
			<DataProviderMock>{children}</DataProviderMock>
		</ThemeContextProvider>
	)
}
