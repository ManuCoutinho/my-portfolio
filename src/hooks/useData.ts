import { DataContext } from 'contexts/DataContext'
import { useContext } from 'react'

export const useData = () => {
	const data = useContext(DataContext)
	return data
}
