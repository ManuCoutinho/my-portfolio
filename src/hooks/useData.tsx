import { DataContext } from 'contexts/DataContext/DataContext'
import { useContext } from 'react'

export const useData = () => {
  const data = useContext(DataContext)
  return data
}
