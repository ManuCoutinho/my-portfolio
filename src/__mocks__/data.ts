import { mockImg } from './images'
export default {
  description: 'a beutiful test',
  site: 'http://www.somesit.com',
  repo: 'http://www.somerepo.com',
  styles: 'styles',
  api: 'mock',
  framework: 'next',
  name: 'project',
  tools: ['tools'],
  id: 'id',
  doc: 'http://www.storybook.com',
  coverDefaultLight: 'img.png',
  coverDefaultDark: '',
  coverAltLight: '',
  coverAltDark: '',
  alt: 'mock img',
  img: mockImg,
  route: '/test'
}
