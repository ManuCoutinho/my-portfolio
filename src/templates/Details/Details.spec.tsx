import React from 'react'
import { screen, waitFor } from '@testing-library/react'
import { render, renderHook } from '__mocks__/customRender'
import mockRouter from 'next-router-mock'
import DetailsTemplate from '.'
import { useData } from 'hooks/useData'
import { customWrapper } from '__mocks__/wrapper'

jest.mock('next/router', () => require('next-router-mock'))
jest.spyOn(React, 'useState')
describe('<DetailsTemplate />', () => {
	beforeEach(() => {
		mockRouter.setCurrentUrl('/?id=81j37ldd-3k344')
		mockRouter.asPath
	})
	it('should render the Details template correctly', async () => {
		const { result } = renderHook(() => useData(), { wrapper: customWrapper() })
		await waitFor(() => expect(result.current.data).not.toBeUndefined())
		const { container } = render(<DetailsTemplate />)
		expect(
			screen.getByRole('heading', { name: /project/i })
		).toBeInTheDocument()
		expect(
			await screen.findByRole('term', { name: /description/ })
		).toBeInTheDocument()
		expect(screen.getByText(/a belezura de test/i)).toBeInTheDocument()
		expect(screen.getByRole('button', { name: /voltar/i })).toBeInTheDocument()
		expect(
			screen.getByRole('link', { name: /documentation/i })
		).toBeInTheDocument()
		expect(screen.getByRole('contentinfo')).toBeInTheDocument()
		expect(container).toMatchSnapshot()
	})
})
