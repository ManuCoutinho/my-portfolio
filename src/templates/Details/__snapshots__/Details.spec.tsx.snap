// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsTemplate /> should render the Details template correctly 1`] = `
<div>
  <main
    class="sc-ec46a235-1 jUdnwh"
  >
    <div
      align="center"
      class="sc-ce9dac95-0 ipIuUD"
      data-testid="stackbox"
      direction="column"
      justify="center"
    >
      <h2
        class="sc-94e9b77b-0 dbunNU"
      >
        project
      </h2>
      <dl
        class="sc-ce9dac95-0 bkDPty"
        data-testid="stackbox"
        direction="column"
        gap="1"
        id="details"
        px="1.5rem"
      >
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="description"
            class="sc-bc329a75-1 ifNFYe"
          >
            Descrição: 
          </dt>
          <dd
            class="sc-f89260d2-0 eyKYlB"
          >
            a belezura de test
          </dd>
        </div>
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="Framework"
            class="sc-bc329a75-1 ifNFYe"
          >
            Framework: 
          </dt>
          <dd
            class="sc-f89260d2-0 eyKYlB"
          >
            next
          </dd>
        </div>
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="tools"
            class="sc-bc329a75-1 ifNFYe"
          >
            Linguagens e Ferramentas:
             
          </dt>
          <dd
            class="sc-f89260d2-0 eyKYlB"
          >
            tools
          </dd>
        </div>
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="style"
            class="sc-bc329a75-1 ifNFYe"
          >
            Estilização: 
          </dt>
          <dd
            class="sc-f89260d2-0 eyKYlB"
          >
            styles
          </dd>
        </div>
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="api"
            class="sc-bc329a75-1 ifNFYe"
          >
            API: 
          </dt>
          <dd
            class="sc-f89260d2-0 eyKYlB"
          >
            mock
          </dd>
        </div>
        <div
          class="sc-bc329a75-0 gbjSaM"
        >
          <dt
            aria-label="available"
            class="sc-bc329a75-1 ifNFYe"
          >
            Disponível em: 
          </dt>
          <a
            class="sc-bc329a75-2 eFEWPK"
            href="http://www.somerepo.com"
            rel="noopener noreferrer"
            target="_blank"
            title="repositorio no github"
          >
            <iconify-icon
              icon="iconoir:github-outline"
              style="font-size: 1.8rem;"
            />
          </a>
          <a
            class="sc-bc329a75-2 eFEWPK"
            href="http://www.somesite.com"
            rel="noopener noreferrer"
            target="_blank"
            title="acessar o project"
          >
            <iconify-icon
              icon="codicon:link-external"
            />
          </a>
          <a
            class="sc-bc329a75-2 eFEWPK"
            href="http://www.storybook.com"
            rel="noopener noreferrer"
            target="_blank"
            title="documentation for project"
          >
            <iconify-icon
              icon="tabler:brand-storybook"
            />
          </a>
        </div>
      </dl>
      <button
        class="sc-66462b9f-0 HjkMD"
        type="button"
      >
        Voltar
      </button>
    </div>
    <footer
      class="sc-31e02b67-0 hUwWXt"
      data-testid="footer"
    >
      <p>
        Copyright © 2023 Todos os direitos reservados. 
      </p>
      <p>
        Designed by 
        <span>
          Manu Coutinho.
        </span>
      </p>
    </footer>
  </main>
</div>
`;
